### PR TITLE
Start formalizing ebnf

### PR DESCRIPTION
**Pull request by caldwecr**
Ready for someone to review the attempt at defining the grammar

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,25 +10,127 @@ MSFL is a context-free language. The context-free grammar is defined below.
...
+
+    binary_op       =   comparison
+                    |   containment ;
+
+    comparison      =   left_curly , comparison_body , { comma , comparison_body } , right_curly ;
+
+    comparison_body =   word , colon , value
+                    |   word , colon , left_curly , comparison_expr , { comma , comparison_expr } , right_curly ;
+
+    comparison_expr =   double_quote , comparison_op , double_quote , colon , value
```
Comment (cymen) [inline README.md]: Missing ending `;`?

File: README.md [hunk 2]
```diff
@@ -10,25 +10,127 @@ MSFL is a context-free language. The context-free grammar is defined below.
...
+
+    comparison_op   =   "lt"
+                    |   "gt"
+                    |   "lte"
+                    |   "gte"
+                    |   "eq"
```
Comment (cymen) [inline README.md]: Missing ending `;`?

File: README.md [hunk 3]
```diff
@@ -10,25 +10,127 @@ MSFL is a context-free language. The context-free grammar is defined below.
...
+
+    containment     =   left_curly , word , colon , left_curly , double_quote , "in" , double_quote , colon , values , right_curly , right_curly
```
Comment (cymen) [inline README.md]: Missing ending `;`?

File: README.md [hunk 4]
```diff
@@ -10,25 +10,127 @@ MSFL is a context-free language. The context-free grammar is defined below.
...
+    or              =   left_curly , double_quote , "or" , double_quote , colon , filters , right_curly ;
+
+    between         =   left_curly , value , colon , start_end , right_curly
+                    |   left_curly , value , colon , between_body , right_curly ;
+
+    between_body    =   left_curly , double_quote , "between" , double_quote , colon , start_end , right_curly ;
+
+    start_end       =   left_curly , start_expr , comma , end_expr , right_curly ;
+
+    start_expr      =   double_quote , "start" , double_quote , colon , range_value
```
Comment (cymen) [inline README.md]: Missing ending `;`?

File: README.md [hunk 5]
```diff
@@ -10,25 +10,127 @@ MSFL is a context-free language. The context-free grammar is defined below.
...
+
+    end_expr        =   double_quote , "end" , double_quote , colon , range_value
```
Comment (cymen) [inline README.md]: Missing ending `;`?

File: README.md [hunk 6]
```diff
@@ -10,25 +10,127 @@ MSFL is a context-free language. The context-free grammar is defined below.
...
+
+    comma           =   ","
+
+    hyphen          =   "-"
+
+    colon           =   ":"
+
+    double_quote    =   '"'
+
+    dot             =   "."
```
Comment (cymen) [inline README.md]: I'm playing with http://www.icosaedro.it/bnf_chk/bnf_chk-on-line.html and it wants a terminating `;` on all of the above (from left_curly on down).

**Comment by cymen** [comment]
Here is an example of EBNF DateTime: http://www.pointillistic.com/ren/ren-ebnf.txt (based on http://www.ietf.org/rfc/rfc3339.txt as it comments). Maybe we can adapt that for our use?

File: README.md (hunk 1 above)
Comment (caldwecr) [inline README.md]: addressed

File: README.md (hunk 2 above)
Comment (caldwecr) [inline README.md]: addressed

File: README.md (hunk 3 above)
Comment (caldwecr) [inline README.md]: addressed

File: README.md (hunk 4 above)
Comment (caldwecr) [inline README.md]: addressed

File: README.md (hunk 5 above)
Comment (caldwecr) [inline README.md]: addressed

**Pull request merged**